processing/src/main/java/org/apache/druid/guice/ExtensionsLoader.java

```diff
@@ -193,9 +193,18 @@ public void initializeExtensionFilesToLoad()
     if (toLoad == null) {
       extensionsToLoad = rootExtensionsDir.listFiles();
     } else {
+      final LinkedHashSet<String> validExtensionsToLoad = new LinkedHashSet<>(toLoad);
+      if (validExtensionsToLoad.remove("druid-multi-stage-query")) {
```

Comment:
Generally it feel a bit awkward having this since configs which are set are not being honored but I guess this might be the cleanest unclean way to support upgrading clusters without manual intervention.